Framework/Muon/inc/MantidMuon/ApplyMuonDetectorGroupPairing.h

```diff
@@ -0,0 +1,120 @@
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+
+class DLLExport ApplyMuonDetectorGroupPairing : public API::Algorithm {
+public:
+  /// (Empty) Constructor
+  ApplyMuonDetectorGroupPairing() : API::Algorithm() {}
+  /// Virtual destructor
+  ~ApplyMuonDetectorGroupPairing() {}
+  /// Algorithm's name
+  const std::string name() const override {
+    return "ApplyMuonDetectorGroupPairing";
+  }
+  /// Algorithm's version
+  int version() const override { return (1); }
+  /// Algorithm's category for identification
+  const std::string category() const override {
+    return "DataHandling\\Grouping;Transforms\\Grouping";
+  }
+  /// Algorithm's summary for identification
+  const std::string summary() const override {
+    return "Perform an asymmetry analysis on two groupings of muon detectors.";
+  }
+  /// See also
+  const std::vector<std::string> seeAlso() const override {
+    return {"MuonProcess", "ApplyMuonDetectorGrouping",
+            "LoadAndApplyMuonDetectorGrouping"};
+  }
+  /// Perform validation of inputs to the algorithm
+  std::map<std::string, std::string> validateInputs() override;
+
+  /// Get the names of the two workspaces in the ADS to pair manually.
+  const std::string
+  getGroupWorkspaceNamesManually(const std::string &groupName,
+                                 const std::string &groupWSName);
+
+  /// Get the name of the workspace to be saved.
+  const std::string getPairWorkspaceName(const std::string &pairName,
+                                         const std::string &groupWSName);
+
+  /// return a workspace for a pair of detector groups, specified in options.
+  API::MatrixWorkspace_sptr
+  createPairWorkspaceManually(API::Workspace_sptr inputWS, const bool noRebin);
+
+  /// Store the input properties in options.
+  Muon::AnalysisOptions getUserInput();
+
+  /// Set MuonProcess properties (input workspace and period properties).
+  void
+  setMuonProcessPeriodProperties(IAlgorithm &alg, API::Workspace_sptr inputWS,
+                                 const Muon::AnalysisOptions &options) const;
+
+  /// Set MuonProcess properties according to the given options.
+  void
+  setMuonProcessAlgorithmProperties(IAlgorithm &alg,
+                                    const Muon::AnalysisOptions &options) const;
+
+  /// Apply the asymmetry calculation to two workspaces.
+  API::MatrixWorkspace_sptr
+  createPairWorkspaceFromGroupWorkspaces(API::MatrixWorkspace_sptr inputWS1,
+                                         API::MatrixWorkspace_sptr inputWS2,
+                                         const double &alpha);
+
+  /// Set grouping properies of MuonProcess
+  void setMuonProcessAlgorithmGroupingProperties(
+      IAlgorithm &alg, const Muon::AnalysisOptions &options) const;
```

Comment:
I think you should you remove the const at the end of the line (and for the one below too)